src/language-js/needs-parens.js

```diff
@@ -121,6 +121,16 @@ function needsParens(path, options) {
     return true;
   }
 
+  // Placeholders used for formatting JS embedded in JS might be substituted by complex
+  // expressions, so the parens around them should not be dropped.
```

Comment:
IMO the `${}` is enough to wrap the expression and extra parentheses are unnecessary.

Reply:
```js
function exprToFn(expr) {
    return Function(/* JS */ ` return ( ${expr} ) + 1 `);
}
```
If Prettier removes the parentheses around `${expr}` (which it would do by default as placeholders are identifiers), this function will break.

Reply:
I see — this is referring to the actual interpolated expression and not the interpolation.